exponentiation_by_squaring/exponentiation_by_squaring.cpp

```diff
@@ -0,0 +1,27 @@
+#include <iostream>
+#include <fstream>
+
+using namespace std;
+
+int x = 1999999973;
+
+int exponentiationBySquaring(long long a, long long b) {
```

Comment:
a and b can be `int`. Then you use 1LL during multiplication. Something like `1LL * a * b`